pyiron_base/storage/helper_functions.py

```diff
@@ -1,8 +1,52 @@
+import numpy as np
 import h5io
+import h5py
+import posixpath
 from pyiron_base.utils.error import retry
+from typing import Union
+import warnings
+
+
+def open_hdf5(filename, mode="r", swmr=False):
+    """
+    Open HDF5 file
+
+    Args:
+        filename (str): Name of the file on disk, or file-like object.  Note: for files created with the 'core' driver,
+                        HDF5 still requires this be non-empty.
+        mode (str): r        Readonly, file must exist (default)
+                    r+       Read/write, file must exist
+                    w        Create file, truncate if exists
+                    w- or x  Create file, fail if exists
+                    a        Read/write if exists, create otherwise
+        swmr (bool): Open the file in SWMR read mode. Only used when mode = 'r'.
+
+    Returns:
+        h5py.File: open HDF5 file object
+    """
+    if swmr and mode != "r":
+        store = h5py.File(name=filename, mode=mode, libver="latest")
+        store.swmr = True
+        return store
+    else:
+        return h5py.File(name=filename, mode=mode, libver="latest", swmr=swmr)
 
 
 def read_hdf5(fname, title="h5io", slash="ignore"):
+    """
+    Read data from HDF5 file
+
+    Args:
+        fname (str): Name of the file on disk, or file-like object.  Note: for files created with the 'core' driver,
+                     HDF5 still requires this be non-empty.
+        title (str): The top-level directory name to use. Typically it is useful to make this your package name,
+                     e.g. ``'mnepython'``.
```

Comment:
```suggestion
        title (str): the HDF5 internal dataset path from which should be read, slashes indicate sub groups
```
Original descriptions feels as if it came from h5io documentation.  For our functions we should document our typical usage of the arguments.

Reply:
Same applies for `write_hdf5` docstring.